editor/src/components/navigator/navigator-item/component-picker.tsx

```diff
@@ -330,67 +530,52 @@ const ComponentPickerComponentSection = React.memo(
           //     currentlySelectedKey={currentlySelectedKey}
           //   />
           // )
-        })}
+
+          return [...acc, ...categoryMarkdown]
+        }, [] as React.ReactNode[])}
       </div>
     )
   },
 )
 
-// FIXME Copy pasted from component-picker-context-menu.tsx
-function iconPropsForIcon(icon: Icon): IcnProps {
-  switch (icon) {
-    case 'column':
-      return {
-        category: 'navigator-element',
-        type: 'flex-column',
-        color: 'white',
-      }
-    case 'row':
-      return {
-        category: 'navigator-element',
-        type: 'flex-row',
-        color: 'white',
-      }
-    case 'regular':
-      return {
-        category: 'navigator-element',
-        type: 'component',
-        color: 'white',
-      }
-    default:
-      assertNever(icon)
+const ICON_TYPE_MAP: Record<Icon, string> = {
+  column: 'flex-column',
+  row: 'flex-row',
+  regular: 'component',
+}
+
+function iconPropsForIcon(icon: Icon, isSelected: boolean = false): IcnProps {
+  return {
+    category: 'navigator-element',
+    color: isSelected ? 'dynamic' : 'white',
+    type: ICON_TYPE_MAP[icon],
   }
 }
 
 interface ComponentPickerOptionProps {
   component: InsertMenuItem
+  id?: string
   onItemClick: (elementToInsert: InsertableComponent) => React.MouseEventHandler
   onItemHover: (elementToInsert: InsertMenuItemValue) => React.MouseEventHandler
   currentlySelectedKey: string | null
 }
 
 const ComponentPickerOption = React.memo((props: ComponentPickerOptionProps) => {
```

Comment:
This is all unused and commented out, but I tried to keep it up to date without adding the map/reduce stuff just incase?